plugin/evm/limitorders/memory_database.go

```diff
@@ -180,7 +165,7 @@ func (db *InMemoryDatabase) UpdateFilledBaseAssetQuantity(quantity *big.Int, ord
 	}
 
 	if limitOrder.BaseAssetQuantity.Cmp(limitOrder.FilledBaseAssetQuantity) == 0 {
-		deleteOrder(db, orderId)
+		limitOrder.FulfilmentBlock = block
```

Comment:
`limitOrder.FulfilmentBlock` should be reset when Accepted state is reverted

Reply:
but accepted block is never reorged

Reply:
Now that we are not deleting the Order and waiting for accepted block to delete it, can we add filter in GetLongOrders and GetShortOrders to skip the order whose fulfimentBlock is > 0